tusker/config.py

```diff
@@ -67,9 +67,14 @@ def __str__(self):
 class MigrationsConfig:
 
     def __init__(self, data):
-        data = ConfigReader(data, 'migrations')
-        self.directory = data.get('directory', str, False)
-        self.filename = data.get('filename', str, False)
+        config = ConfigReader(data, 'migrations')
+        self.directory = config.get('directory', str, False)
+
+        if isinstance(data['filename'], str):
```

Comment:
This is not correct. The `ConfigReader` class only exposes a `get` method. You should call `config.get('filename', (str, list))` instead. While we're at it how about adding validation that a list of strings is passed:

```python
self.filename = config.get('filename', (str, list))
if isinstance(self.filename, str):
    self.filename = [self.filename]
else:
    if not all(isinstance(x) for x in self.filename):
        raise ConfigError.invalid(name, 'Not a list of strings {}'.format(self.filename))
```

Maybe even better would be to add a `get_list` method to the `ConfigReader` which does implement this behavior. That way it would be a very simple change to enable this feature for the schema filename, too.

Reply:
Oh wait. You're accessing `data` directly. I take that back. Your code is fine. Even though I find it a bit strange to first pass the "ownership" of the data to the `ConfigReader` and then access `data` directly.

Reply:
Didn't know specifying two types was possible.  Makes it cleaner to do it this way, I think.

```
self.filename = config.get('filename', (str, list))
```